conf/route_control.py

```diff
@@ -3,528 +3,884 @@
+                    module,
+                    ogate,
+                    igate,
+                    next_module,
+                    SLEEP_S,
+                )
+                time.sleep(SLEEP_S)
             else:
-                bess.resume_all()
-                print('Module {} deletion failure.'.format(update_module))
-                return
-                #raise Exception('Module {} deletion failure.'.
-                #                format(update_module))
-
-            print('Module {} destroyed'.format(update_module))
-
-            # Delete entry from the neighbor cache
-            del neighborcache[item.neighbor_ip]
-            print('Deleting item from neighborcache')
-            del neighbor_exists
-        else:
-            print('Route count for {}  decremented to {}'.format(
-                item.neighbor_ip, neighbor_exists.route_count))
-            neighborcache[item.neighbor_ip] = neighbor_exists
-    else:
-        print('Neighbor {} does not exist'.format(item.neighbor_ip))
-
-
-def probe_addr(item, src_mac):
-    # Store entry if entry does not exist in ARP cache
-    arpcache[item.neighbor_ip] = item
-    print('Adding entry {} in arp probe table'.format(item))
-
-    try:
-        ipb = ipaddress.ip_address(item.neighbor_ip)
-        if isinstance(ipb, ipaddress.IPv4Address):
-          print("The IP address {} is valid ipv4 address".format(ipb))
+                logger.info(
+                    "Module %s:%i->%i/%s connected",
+                    module,
+                    ogate,
+                    igate,
+                    next_module,
+                )
+                break
+            finally:
+                self.bess.resume_all()
         else:
-          print("The IP address {} is valid ipv6 address. Ignore ".format(ipb))
-          return
-    except:
-        print("The IP address {} is invalid".format(item.neighbor_ip))
-        return
-    # Probe ARP request by sending ping
-    send_ping(item.neighbor_ip)
-
-    # Probe ARP request
-    ##send_arp(neighbor_ip, src_mac, item.iface)
-
-
-def parse_new_route(msg):
-    item = NeighborEntry()
-    # Fetch prefix_len
-    item.prefix_len = msg['dst_len']
-    # Default route
-    if item.prefix_len == 0:
-        item.iprange = '0.0.0.0'
-
-    for att in msg['attrs']:
-        if 'RTA_DST' in att:
-            # Fetch IP range
-            # ('RTA_DST', iprange)
-            item.iprange = att[1]
-        if 'RTA_GATEWAY' in att:
-            # Fetch gateway MAC address
-            # ('RTA_GATEWAY', neighbor_ip)
-            item.neighbor_ip = att[1]
-            _mac = fetch_mac(att[1])
-            if not _mac:
-                gateway_mac = 0
+            raise Exception(
+                "BESS module connection ({}:{}->{}:{}) failure.".format(
+                    module, ogate, igate, next_module
+                )
+            )
+
+    def del_module(self, module_name) -> None:
```

Comment:
We use "create", "link", "update", why not use "delete" (instead of "del")

Reply:
done